lib/Storage/CacheableFlysystemAdapter.php

```diff
@@ -177,4 +177,12 @@ public function filetype($path) {
 		$info = $this->getFlysystemMetadata($path);
 		return $info['type'];
 	}
+
+	/**
+	* {@inheritdoc}
+	*/
+	public function file_exists($path) {
+		$info = $this->getFlysystemMetadata($path);
```

Comment:
just wondering, I see that `stat()` above does the same call but doesn't check for null. Would be good to add a null check there

Reply:
I think may be these functions are called after `file_exists` so that they don't return `null|false`
Because FTP adapter is also doing the same thing
https://github.com/owncloud/files_external_ftp/blob/master/lib/Storage/FTP.php#L145

Reply:
Would still be safer to add a proper check instead of relying on the caller calling in the right order. It is likely that the developer who wrote the FTP adapter didn't notice either.

Can you adjust ? Better be safe

Reply:
Hmm sounds logical
I have added some checks to tackle this